tests/testthat/test_all_open.R

```diff
@@ -0,0 +1,23 @@
+library(testthat)
+context("all open intervals\n")
+library(iregnet)
+
+set.seed(1)
+toy.features <- matrix(rnorm(10), 5, 2)
+toy.targets <- rbind(
+  c(1, 2),
+  c(-Inf, 5),
+  c(1, Inf),
+  c(-Inf, 3),
+  c(-Inf, 4))
+
+test_that("iregnet works with at least one closed interval", {
+  fit <- iregnet(toy.features, toy.targets)
+  expect_true(inherits(fit, "iregnet"))
+})
+
+test_that("iregnet works for all open intervals", {
+  fit <- iregnet(toy.features[-1,], toy.targets[-1,])
+  expect_true(inherits(fit, "iregnet"))
```

Comment:
here is a test that fails with the current code -- a simple data set with four open intervals for labels, "NANs produced" error somewhere in the C++ code, but it should return a valid model.